doc/nrf/ug_matter_hw_requirements.rst

```diff
@@ -162,3 +166,302 @@ Values are provided in kilobytes (KB).
       +------------------------------------------------------------+---------------+-------------------+----------------+------------+-------------+-------------+
 
 ..
+
+.. _ug_matter_hw_requirements_layouts:
+
+Matter memory layouts
+*********************
+
+The following sections show how the conceptual :ref:`ug_matter_overview_architecture` translates to actual memory maps for each of the available :ref:`ug_matter_overview_architecture_integration_designs`.
+The memory values match `RAM and flash memory requirements`_ listed above.
+
+The following tabs show the memory maps for the development kits supported by the Matter protocol, including two memory maps for the :ref:`matter_weather_station_app`, which uses Nordic Thingy:53.
+
+.. tabs::
+
+   .. tab:: nRF52840 DK
+
+      The following memory map is valid for Matter applications running on the :ref:`nRF52840 DK <gs_programming_board_names>` (:ref:`nrf52840dk_nrf52840 <zephyr:nrf52840dk_nrf52840>`).
+
+      .. table:
+
+         +--------------------------------+--------------------------+----------------+-------------------+---------------------+-----------------+-----------------+
+         | Memory region (incl. size)     | Partition                | Offset         | Size              | Partition elements  | Element offset  | Element size    |
+         +================================+==========================+================+===================+=====================+=================+=================+
+         | External flash (QSPI)          | mcuboot_secondary        | 0kB (0x0)      | 976kB (0xf4000)   | -                   | -               | -               |
+         + (0x800000 = 8192kB)            +--------------------------+----------------+-------------------+---------------------+-----------------+-----------------+
+         |                                | external_flash           | 976kB (0xf4000)| 7216kB (0x70c000) | -                   | -               | -               |
+         +--------------------------------+--------------------------+----------------+-------------------+---------------------+-----------------+-----------------+
+         | Flash primary                  | mcuboot                  | 0kB (0x0)      | 28kB (0x7000)     | -                   | -               | -               |
```

Comment:
I think we use the term "internal flash" in other docs, so maybe it would be more understandable to call it like that here.